src/PolygonDraw/PolygonVertex.tsx

```diff
@@ -21,93 +19,97 @@ export interface Props {
     onDragEnd: () => void;
 }
 
-export interface State {
-    isHovered: boolean;
-    isDragged: boolean;
-    latLng: LatLng;
-}
-
-export class PolygonVertex extends React.Component<Props, State> {
-    private circleMarkerElement: CircleMarker;
-    state = {
-        isHovered: false,
-        isDragged: false,
-        latLng: new LatLng(0, 0),
-    };
+export const PolygonVertex: FC<Props> = ({
+    coordinate,
+    isSelected,
+    index,
+    onClick,
+    onDrag,
+    onDragEnd,
+    onDragStart,
+}) => {
+    const [isHovered, setIsHovered] = useState(false);
+    const [isDragged, setIsDragged] = useState(false);
+    const [latLng, setLatLng] = useState(new LatLng(0, 0));
 
-    static getDerivedStateFromProps = (props: Props, state: State) => {
-        return state.isDragged ? state : { ...state, latLng: createLeafletLatLngFromCoordinate(props.coordinate) };
-    };
+    const circleMarkerRef = useRef<CircleMarker>();
 
-    componentDidMount() {
-        if (this.circleMarkerElement) {
-            this.circleMarkerElement.on('dragstart', this.handleDragStart);
-            this.circleMarkerElement.on('drag', this.handleDrag);
-            this.circleMarkerElement.on('dragend', this.handleDragEnd);
+    const setCircleMarkerRef = useCallback((ref: CircleMarker | null) => {
+        if (ref) {
+            circleMarkerRef.current = ref;
         }
-    }
+    }, []);
 
-    componentWillUnmount() {
-        if (this.circleMarkerElement) {
-            this.circleMarkerElement.off('dragstart', this.handleDragStart);
-            this.circleMarkerElement.off('drag', this.handleDrag);
-            this.circleMarkerElement.off('dragend', this.handleDragEnd);
+    useEffect(() => {
```

Comment:
(optional, personal preference) i think we could remove this `useEffect` by adding `createLeafletLatLngFromCoordinate` as the state init value and in the `handleDragEnd` method

Reply:
This is unfortunately not working as the coordinates need to stay as reactive dependency that the marker needs to know about. Without the effect the marker stays at an old position e.g. when the undo action is used.